src/WebDAVAdapter.php

```diff
@@ -126,6 +135,10 @@ public function read($path)
      */
     public function write($path, $contents, Config $config)
     {
+        if (!$this->createDir(dirname($path), $config)) {
```

Comment:
Dirname returns `.` when there's no parent directory, the `Util::dirname` normalizes this.